src/wix/WixToolset.Core.Burn/Bundles/BurnReader.cs

```diff
@@ -142,6 +142,69 @@ public bool ExtractUXContainer(string outputDirectory, string tempDirectory)
             return true;
         }
 
+        /// <summary>
+        /// Extracts detached containers to the output directory.
+        /// </summary>
+        /// <param name="outputDirectory">Directory to write extracted files to.</param>
+        /// <param name="tempDirectory">Scratch directory.</param>
+        /// <param name="uxOutputDirectory">UX extraction folder. If null or empty, a UX folder will be created within tempDirectory</param>
+        /// <returns>True if successful, false otherwise</returns>
+        public bool ExtractDetachedContainers(string outputDirectory, string tempDirectory, string uxOutputDirectory)
+        {
+            if (string.IsNullOrEmpty(uxOutputDirectory))
+            {
+                uxOutputDirectory = Path.Combine(tempDirectory, "UX", "Final");
```

Comment:
`UX` is a deprecated term. I know this is a temp folder, but let's not spread `UX` in more (possibly) user visible locations and use `BA` instead.